pwa/src/offline-interface/offline-interface.js

```diff
@@ -58,44 +59,88 @@ export class OfflineInterface {
             return null
         }
 
-        function onUpdate(registration) {
-            if (!promptUpdate) {
+        // This event emitter helps coordinate with service worker messages
+        this.offlineEvents = new EventEmitter()
+
+        // Receives messages from service worker and forwards to event emitter
+        const handleSWMessage = event => {
+            if (!event.data) {
                 return
             }
+            const { type, payload } = event.data
+            this.offlineEvents.emit(type, payload)
+        }
+        navigator.serviceWorker.addEventListener('message', handleSWMessage)
```

Comment:
This is just moved up from below so the 'update' logic can send and receive messages to get the number of clients open